bigchaindb/block.py

```diff
@@ -229,3 +232,89 @@ def _start(self):
         p_write.start()
         p_delete.start()
 
+
+class BacklogDeleteRevert(Block):
+
+    def __init__(self, q_backlog_delete):
+        # invalid transactions can stay deleted
+        self.q_tx_to_validate = q_backlog_delete
+        self.q_tx_validated = mp.Queue()
+        self.q_transaction_to_revert = mp.Queue()
+        self.q_tx_delete = mp.Queue()
+
+        self.monitor = Monitor()
+
+    def locate_transactions(self):
+        """
+        Determine if a deleted transaction has made it into a block
+        """
+        # create bigchain instance
+        b = Bigchain()
+
+        while True:
+            tx = self.q_tx_validated.get()
+
+            # poison pill
+            if tx == 'stop':
+                self.q_tx_delete.put('stop')
+                self.q_transaction_to_revert.put('stop')
+                return
+
+            # check if tx is in a (valid) block
+            validity = b.get_blocks_status_containing_tx(tx['id'])
+
+            if validity and list(validity.values()).count(Bigchain.BLOCK_VALID) == 1:
+                # tx made it into a block, and can safely be deleted
+                self.q_tx_delete.put(tx['id'])
```

Comment:
It's there because `BacklogDeleteRevert` is a derived class of block, so I didn't have to rewrite the `validate_transactions`. It expects a queue to take invalid transactions, but yes it's functionally useless.